presto-native-execution/presto_cpp/main/CMakeLists.txt

```diff
@@ -20,6 +20,7 @@ add_library(
   Announcer.cpp
   CPUMon.cpp
   CoordinatorDiscoverer.cpp
+  LinuxMemoryChecker.cpp
```

Comment:
We should avoid config options if possible. The LinuxMemoryChecker is a smallish code.
Let's always build it if the platform is Linux.
```
if(LINUX) 
  target_sources(presto_server_lib LinuxMemoryChecker.cpp)
endif()
```

Reply:
Yes, that works too. This way we won't need the build flag - this would get rid of this - or are you also referring to the option to enable this besides the build flag?

Reply:
@majetideepak I see an example of this used instead in /presto/presto-native-execution/velox/CMakeLists.txt:
```
if(UNIX AND NOT APPLE)
  # linker flags, -export-dynamic for rtti
  add_link_options("-Wl,-export-dynamic")
endif()
```

I think I will use this instead of `if(LINUX)` as I don't see that used anywhere in the codebase

Reply:
> or are you also referring to the option to enable this besides the build flag?

@czentgr the config option should stay. The build option can go away.

> if(UNIX AND NOT APPLE)

@minhancao this works!